Fix normalize_screen usage with the `writeelegant` option

st-walker opened the pull request:
With the option `writeelegant` Particles, pCentral and Charge are additionally written to the screen
block.  As these are all floats and not arrays, this crashes when trying to call `.size` on
it.  Solution is to loop over arr_keys, same as used in normalize_tout (par_keys).

st-walker committed:
Fix normalize_screen usage with the `writelegant` option

With writelegant Particles, pCentral and Charge are additionally written to the screen
block, as these are all floats and not arrays, this crashes when trying to call .size on
it.  Solution is to loop over arr_keys, same as used in normalize_tout (par_keys).

electronsandstuff commented:
Thanks for the contribution!  I'll try to take a look at it over the weekend.

electronsandstuff commented:
Didn't realize this was a single line PR.  Looks good to me, but let me add a test case so we can check for the problem with future changes and also bump the version number.

electronsandstuff committed:
go to old code to write unit test that fails on issue

electronsandstuff committed:
add unit test that fails on bug

electronsandstuff committed:
add suggested fix and modify to check if key in screen

not all screens passed to normalize_screen have all elements of arr_keys and to pass all unit tests, the additional check is required

electronsandstuff committed:
version bump

electronsandstuff commented:
Ok, the only change I ended up making was an additional check to make sure the screen contains the array element before we ask for its size.  Some file don't include all of them and it was setting off some unit tests.  I'll go ahead and merge now.
